scheduler/services/resource/base.py

```diff
@@ -575,12 +598,99 @@ def none_to_str(value) -> str:
+        Parse Engineering task that block moments or the entire night.
+        Each twilight is calculated using lucupy.sky, some discrepancies might affect.
+        """
+        pattern = r'(\[.*\])'
+        # Ignore GS until the file is created.
+        if site is Site.GS:
+            return
+
+        with open(os.path.join(self._path, to_file), 'r') as file:
+            for line_num, line in enumerate(file):
+                # Find pattern to keep bracket comments not split.
+                match = re.search(pattern, line)
+                if match:
+                    comment = match.group(1)
+                    rest_of_line = re.sub(pattern, '', line)
+                    eng_date, start_time, end_time = rest_of_line.strip().split()
+                    #  Single date for key
+                    just_date = datetime.strptime(eng_date, '%Y-%m-%d')
+                    # Time day in jd to calculate twilight
+                    time = Time(just_date)
+                    _, _, _, even_12twi, morn_12twi, _, _ = night_events(time,
+                                                                         site.location,
+                                                                         site.timezone)
+
+                    # Handle twilight
+                    if start_time == 'twi':
+                        start_date = even_12twi.datetime
+                        end_date = (morn_12twi.datetime if end_time == 'twi' else
+                                    datetime.strptime(f'{eng_date} {end_time}', '%Y-%m-%d %H:%M'))
+                    else:
+                        start_date = datetime.strptime(f'{eng_date} {start_time}', '%Y-%m-%d %H:%M')
+                        end_date = (morn_12twi.datetime if end_time == 'twi'
+                                    else datetime.strptime(f'{eng_date} {end_time}', '%Y-%m-%d %H:%M'))
+
+                    time_loss = end_date - start_date
+                    eng_task = EngTask(start=start_date,
+                                       end=end_date,
+                                       site=site,
+                                       time_loss=time_loss,
+                                       reason=comment)
+
+                    if just_date in self._eng_task[site]:
+                        self._eng_task[site][just_date].add(eng_task)
+                    else:
+                        self._eng_task[site][just_date] = {eng_task}
+                else:
+                    raise ValueError(f'Pattern not found. Format error on Eng Task file at line {line_num}')
+
+    def _parse_faults_file(self, site: Site, to_file: str) -> None:
+        """Parse faults from files.
+        This is purposeful left non-private as might be used with incoming files from
+        the React app.
+        """
+        # Files contains this repetitive string in each timestamp, if we need them
+        # we could add them as constants.
+        ts_clean = ' 04:00' if site is Site.GS else ' 10:00'
```

Comment:
Instead of doing this, can we just preprocess the files to not have this redundancy? Do a find / replace and just get rid of it?

Reply:
I'm wasn't sure if it was going to be needed or not.

Reply:
I think we need a standardized file format, and then we take the existing files and make them match the standardized format.

We send the standard out to science staff, and use it internally if we are accepting text files for these values.

So, we:
1. Create the standard
2. Make the existing files match the standard (with a script to just read and fix them, or by fixing them by hand, but this should not be run by the Scheduler)
3. Use the standardized output files in the Scheduler